Make `parity_synth` importable from `passes` module (#2553)
PennyLaneAI/catalyst: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

* Combining ``GlobalPhase`` operations into one single operation is now possible with the

>   :func:`catalyst.passes.combine_global_phases` pass.

Author: Could be wrong but I don't think that this function was documented anywhere before 😱

  [(#2553)](https://github.com/PennyLaneAI/catalyst/pull/2553)

  ```python
  import pennylane as qml
  import catalyst

  @qml.qjit(capture=True)
  @catalyst.passes.combine_global_phases
  @qml.qnode(qml.device("lightning.qubit", wires=5))
  def circuit():
      qml.GlobalPhase(0)
      qml.GlobalPhase(1)
      qml.GlobalPhase(2)
      qml.GlobalPhase(3)
      qml.GlobalPhase(4)
      return qml.state()
  ```

  ```pycon

>   >>> print(qml.specs(circuit, level=2)())

Reviewer: ````suggestion
  import pennylane as qp
  import catalyst

  @qp.qjit(capture=True)
  @catalyst.passes.combine_global_phases
  @qp.qnode(qml.device("lightning.qubit", wires=5))
  def circuit():
      qp.GlobalPhase(0)
      qp.GlobalPhase(1)
      qp.GlobalPhase(2)
      qp.GlobalPhase(3)
      qp.GlobalPhase(4)
      return qp.state()
  ```

  ```pycon
  >>> print(qp.specs(circuit, level=2)())
````

<h3>Improvements 🛠</h3>

* The :func:`~.passes.parity_synth` can now be invoked from the ``passes`` module.
  [(#2553)](https://github.com/PennyLaneAI/catalyst/pull/2553)

  ```python
  import pennylane as qml
  import catalyst

  dev = qml.device("lightning.qubit", wires=2)

  @qml.qjit(capture=True)
  @catalyst.passes.parity_synth
  @qml.qnode(dev)
  def circuit(x: float, y: float, z: float):
      qml.CNOT((0, 1))
      qml.RZ(x, 1)
      qml.CNOT((0, 1))
      qml.RX(y, 1)
      qml.CNOT((1, 0))
      qml.RZ(z, 1)
      qml.CNOT((1, 0))
      return qml.state()
  ```

  ```pycon

>   >>> print(qml.specs(circuit)(0.52, 0.12, 0.2))

Reviewer: ````suggestion
  import pennylane as qp
  import catalyst

  dev = qp.device("lightning.qubit", wires=2)

  @qp.qjit(capture=True)
  @catalyst.passes.parity_synth
  @qp.qnode(dev)
  def circuit(x: float, y: float, z: float):
      qp.CNOT((0, 1))
      qp.RZ(x, 1)
      qp.CNOT((0, 1))
      qp.RX(y, 1)
      qp.CNOT((1, 0))
      qp.RZ(z, 1)
      qp.CNOT((1, 0))
      return qp.state()
  ```

  ```pycon
  >>> print(qp.specs(circuit)(0.52, 0.12, 0.2))
````

  Device: lightning.qubit
  Device wires: 2
  Shots: Shots(total=None)
  Level: device

  Wire allocations: 2
  Total gates: 5
  Gate counts:
  - RX: 1
  - RZ: 2
  - CNOT: 2
  Measurements:
  - state(all wires): 1
  Depth: 5
  ```

  Note as well that this compilation pass used to be named ``parity_synth_pass``.

* ``ResourceAnalysis`` and ``RegisterDecompRuleResource`` passes now record the number of classical
  parameters for each gate alongside the wire count. The operation key format changes from
  `"GateName(nWires)"` to `"GateName(nWires,nParams)"`.
  [(#2755)](https://github.com/PennyLaneAI/catalyst/pull/2755)

> * `qml.for_loop` now supports dynamic shapes with program capture `qjit(capture=True)`.

Reviewer: ```suggestion
* `qp.for_loop` now supports dynamic shapes with program capture `qjit(capture=True)`.
```

Author: :screaming:

<h3>Breaking changes 💔</h3>

* The ``catalyst.python_interface.transforms.parity_synth_pass`` and
  ``catalyst.python_interface.transforms.combine_global_phases_pass`` transforms have been renamed
  to ``catalyst.python_interface.transforms.parity_synth`` and
  ``catalyst.python_interface.transforms.combine_global_phases``, respectively.

>   [(#2553)](https://github.com/PennyLaneAI/catalyst/pull/2553)

Author: TODO

<h3>Breaking changes 💔</h3>

* The ``catalyst.python_interface.transforms.parity_synth_pass`` and
  ``catalyst.python_interface.transforms.combine_global_phases_pass`` transforms have been renamed
  to ``catalyst.python_interface.transforms.parity_synth`` and

>   ``catalyst.python_interface.transforms.combine_global_phases``, respectively.

Reviewer: The xDSL combine_global_phases` doesn't exist anymore

Author: ah yes good catch I forgot to update this!

```suggestion
* The ``catalyst.python_interface.transforms.parity_synth_pass`` transform has been renamed to ``catalyst.python_interface.transforms.parity_synth``.
```
